exp-player/addon/components/exp-player.js

```diff
@@ -1,59 +1,40 @@
 import Ember from 'ember';
 import layout from '../templates/components/exp-player';
+import parseExperiment from '../utils/parse-experiment';
 
 export default Ember.Component.extend({
     layout: layout,
-    store: Ember.inject.service('store'),
 
+    experiment: null, // Experiment model
     frames: null,
-    frameIndex: null,
-    _last: null,
-    ctx: {
-        data: {}
+
+    frameIndex: null,  // Index of the currently active frame
+
+    expData: {},  // Temporarily store data collected until we sent to server at end
+
+    init: function() {
+        this._super(...arguments);
+        this.set('frameIndex', 0);
+        var frameConfigs = parseExperiment(this.get('experiment.structure'));
```

Comment:
Indeed, that's one of the "questions for reviewers" in the companion PR: as it stands this doesn't provide a great way.

>  What is the best way to track which path a user went down? Currently the only way is to track frame IDs in the final payload.

...eg, the keys in `expData`.

That said, we could easily extract the sequence of IDs from the sequence of `frameConfig` objects. Would you be comfortable storing an ordered array of frame IDs (specifying the experiment sequence) on the `session` model? 

Is there a more natural way to store frames that would better fit the experimenter's workflow?